### PR TITLE
Add extra field support

### PR DESCRIPTION
**Pull request by inorick**
This change adds initial read and write support for extra fields of ZIP files.

### CODE REVIEW

File: src/read.rs [hunk 1]
```diff
@@ -349,7 +349,7 @@ impl<R: Read + io::Seek> ZipArchive<R> {
     }
 
     /// Search for a file entry by name
-    pub fn by_name<'a>(&'a mut self, name: &str) -> ZipResult<ZipFile<'a>> {
+    pub fn by_name(&mut self, name: &str) -> ZipResult<ZipFile> {
```
Comment (Plecra) [inline src/read.rs:352]: This will unify the lifetimes of `self` and `name`, changing the API. Did clippy recommend this change?

**Comment by Plecra** [comment]
Hiya! Thanks for the PR. Frustratingly, I haven't had the time to properly review PRs in the last couple months due to other commitments. I'm hoping to find a weekend to set aside for merging some of the changes soon.

It'd be great if you could explain how this differs from #200, which I'm almost ready to merge.

**Comment by inorick** [comment]
I had a look at #200 and it looks like a more complete implementation that also handles ZIP64. I did not test it but from the looks of it I would prefer it.

**Comment by inorick** [comment]
Closing in favour of #200

**Pull request closed without merging**